e2e/e2e.go

```diff
@@ -2,41 +2,101 @@
 package e2e
 
 import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	aoapis "github.com/openshift/addon-operator/apis"
+	"github.com/stretchr/testify/require"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
 
-	aoapis "github.com/openshift/addon-operator/apis"
+const (
+	relativeConfigDeployPath = "../../config/deploy"
 )
 
 var (
 	// Client pointing to the e2e test cluster.
-	c      client.Client
-	scheme = runtime.NewScheme()
+	Client client.Client
+	Scheme = runtime.NewScheme()
+
+	// Path to the deployment configuration directory.
+	PathConfigDeploy string
 )
 
 func init() {
-	err := clientgoscheme.AddToScheme(scheme)
+	// Client/Scheme setup.
+	err := clientgoscheme.AddToScheme(Scheme)
 	if err != nil {
 		panic(err)
 	}
 
-	err = aoapis.AddToScheme(scheme)
+	err = aoapis.AddToScheme(Scheme)
 	if err != nil {
 		panic(err)
 	}
 
-	err = apiextensionsv1.AddToScheme(scheme)
+	err = apiextensionsv1.AddToScheme(Scheme)
 	if err != nil {
 		panic(err)
 	}
 
-	c, err = client.New(ctrl.GetConfigOrDie(), client.Options{
-		Scheme: scheme,
+	Client, err = client.New(ctrl.GetConfigOrDie(), client.Options{
+		Scheme: Scheme,
 	})
 	if err != nil {
 		panic(err)
 	}
+
+	// Paths
+	PathConfigDeploy, err = filepath.Abs(relativeConfigDeployPath)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Load all k8s objects from .yaml files in config/deploy.
+// File/Object order is preserved.
+func LoadObjectsFromDeploymentFiles(t *testing.T) []unstructured.Unstructured {
+	configDeploy, err := os.Open(PathConfigDeploy)
+	require.NoError(t, err)
+
+	files, err := configDeploy.Readdir(-1)
+	require.NoError(t, err)
+
+	var objects []unstructured.Unstructured
+	for _, f := range files {
+		if f.IsDir() {
+			continue
```

Comment:
I think this should be a call to `panic(...)` to ensure that no one puts a folder there and expects it to work.

Reply:
That could also be a `.something` folder or just other configuration.
I don't see a need to validate the folder structure in here.

Reply:
Fair argument!